src/intTest/java/com/box/sdk/BoxSignRequestIT.java

```diff
@@ -32,89 +29,100 @@ public static void tearDown() {
+
+            // todo: get signer by role type. Using index=1 is fragile, as order may not be guaranteed.
+            //signer at index 0 has role=final_copy_reader
+            //signer at index 1 has role=signer
+            BoxSignRequestSigner signerCreate = signRequestInfoCreate.getSigners().get(1);
+
+            // Test Create
+            assertNotNull(signRequestInfoCreate.getPrepareUrl());
+            assertEquals(file.getID(), fileInfoCreate.getID());
+            assertEquals(signerEmail, signerCreate.getEmail());
+            assertNotNull(signRequestInfoCreate.getID());
+
+            // Do Get by ID
+            BoxSignRequest signRequestGetByID = new BoxSignRequest(api, signRequestIdCreate);
+            BoxSignRequest.Info signRequestInfoGetByID = signRequestGetByID.getInfo();
+            BoxFile.Info fileInfo = signRequestInfoGetByID.getSourceFiles().get(0);
+
+            // Todo: get signer by role type. Using index=1 is fragile, as order may not be guaranteed.
+            //signer at index 0 has role=final_copy_reader
+            //signer at index 1 has role=signer
+            BoxSignRequestSigner signer = signRequestInfoGetByID.getSigners().get(1);
+
+            // Test Get by ID
+            assertEquals(file.getID(), fileInfo.getID());
+            assertEquals(signerEmail, signer.getEmail());
+            assertEquals(signRequestIdCreate, signRequestInfoGetByID.getID());
+
+            // Do Get All
+            Iterable<BoxSignRequest.Info> signRequestsGetAll = BoxSignRequest.getAll(api);
+
+            // Test Get All
+            assertTrue(signRequestsGetAll.iterator().hasNext());
+
+            // Do Cancel
+            // Cancel will fail if it's too soon after creation
+            Thread.sleep(3000);
+            signRequestInfoCancel = signRequestGetByID.cancel();
+            BoxSignRequest signRequestGetByIDAfterCancel = new BoxSignRequest(api, signRequestIdCreate);
+            BoxSignRequest.Info signRequestInfoAfterCancel = signRequestGetByIDAfterCancel.getInfo();
+            BoxSignRequest.BoxSignRequestStatus signRequestStatusAfterCancel = signRequestInfoAfterCancel.getStatus();
+
+            // Test Cancel
+            assertEquals(BoxSignRequest.BoxSignRequestStatus.Cancelled, signRequestInfoCancel.getStatus());
+            assertEquals(BoxSignRequest.BoxSignRequestStatus.Cancelled, signRequestStatusAfterCancel);
+        } finally {
+            if (signRequestInfoCancel != null) {
+                // Clean up
+                List<BoxFile.Info> signRequestFiles = signRequestInfoCancel.getSignFiles().getFiles();
+                for (BoxFile.Info signRequestFile : signRequestFiles) {
+                    BoxFile fileToDelete = new BoxFile(api, signRequestFile.getID());
+                    fileToDelete.delete();
+                }
+            }
+
+            // Deleting the folder before the above file will cause sign request 500s on future calls
+            Thread.sleep(3000);
```

Comment:
Why do we need to wait for 3000ms here? Is it because we need to wait until the files are deleted?

Reply:
You are right we do not need to wait those 3s